### ISSUE TITLE
Enhance the current admin dashboard.

### ISSUE THREAD

**Issue by usha-madithati** [usha-madithati/esmart.github.io#327]
Needs to enhance the new functionalities like adding admin acces to user accounts. We needs to have a admin dashboard where admin can know each and every activity of users

**Comment by MinavKaria** [comment]
What all activities do you want to be shown on admin dashboard? In how much time (Max days) do you expect to complete this task?

**Comment by usha-madithati** [comment]
> @usha-madithati What all activities do you want to be shown on admin dashboard? In how much time (Max days) do you expect to complete this task? 

I have shared in the description right . Those are all the activities we wants to have . As it involves ui and even backend we will assign you 5 days of time.are you ok with that ?

**Comment by Ansh101112** [comment]
Ma'am I also want to work on this issue as I have built the admin dashboard and knowledge of the enhancement structure that needs to be implemented here.

**Comment by MinavKaria** [comment]
@usha-madithati , I can do this,
So basically we have to,
- add middleware
- add activity model schema
- fetch the activity on frontend admin page and display as table
- list of all the user with toggle to give user access, so add a parameter in the user schema of adminAccess: [true|false]

Is that right and enough ? Can I use MUI ? Or Existing packages only ?

@usha-madithati assign it to me

**Comment by usha-madithati** [comment]
> @usha-madithati , I can do this,
> So basically we have to,
> - add middleware
> - add activity model schema
> - fetch the activity on frontend admin page and display as table
> - list of all the user with toggle to give user access, so add a parameter in the user schema of adminAccess: [true|false]
> 
> Is that right and enough ? Can I use MUI ? Or Existing packages only ?
> 
> @usha-madithati assign it to me 

Yeah ok . We have other contributor to0 so name a deadline we will see if you can submit the work before him we will assign you .

**Comment by MinavKaria** [comment]
2-3 days from today, Bcoz I'll be working from tomorrow.

**Comment by usha-madithati** [comment]
Ok drop your discord details . I will contact you there

**Comment by MinavKaria** [comment]
username: mrcoleno
